tests/linalg_test.py

```
@@ -2115,6 +2122,69 @@ def testToeplitzConstructionWithKnownCases(self):
       [2, 1, 4, 5],
       [3, 2, 1, 4]], dtype=np.float32))
 
+  @jtu.sample_product(
+    shape=[(2, 3), (4, 6), (5, 7), (100, 300)],
+    dtype = float_types + complex_types,
+    method = ["schur", "eigen"]
+  )
+  @jtu.run_on_devices("cpu", "gpu")
+  def test_solve_sylvester(self, shape, dtype, method):
+    if jtu.test_device_matches(["gpu"]) and method == "schur":
+      self.skipTest("Schur not supported on GPU.")
+
+    tol = {np.float32: 3e-2, np.complex64: 3e-2, np.complex128: 1e-12}
```

It looks like there's still a failure at around the `1E-10` level.

---

hmm, anyway I can test this locally in bulk? I ran this test at least 20 times manually and it was passing at `1e-12` level.

---

@jakevdp, I updated the `tol` to `1E-9` level for `complex128` types. I ran the `complex128` test 1000 times and all randomly generated matrices passed at that `tol` level.

---

I think if the solver is useful for complex types, we should test the regions in which it's useful. If the solver is not useful for complex types, we should raise an error when the user passes a complex type. If there's any gray area, we should test the regime that makes sense for testing, but also add a documentation note about any regimes where things may go wrong.

---

Yes, I agree with you. It seems that the solver (similar to Scipy) has limitations when the size of the matrix increases. I am finding that when ether axis of the matrix increases beyond ~120 the solver fails to produce a good solution for complex128 types. I updated the tests to remain in that region of matrix sizes and I added a note on this degradation of results in the notes section of the docstring.